src/substrait/utils.py

```diff
@@ -0,0 +1,44 @@
+import substrait.gen.proto.type_pb2 as stp
+import substrait.gen.proto.extensions.extensions_pb2 as ste
+from typing import Iterable
+
+def type_num_names(typ: stp.Type):
+    kind = typ.WhichOneof("kind")
+    if kind == "struct":
+        lengths = [type_num_names(t) for t in typ.struct.types]
+        return sum(lengths) + 1
+    elif kind == "list":
+        return type_num_names(typ.list.type)
+    elif kind == "map":
+        return type_num_names(typ.map.key) + type_num_names(typ.map.value)
+    else:
+        return 1
+
+def merge_extension_uris(*extension_uris: Iterable[ste.SimpleExtensionURI]):
```

Comment:
A comment explaining what merge does (maintains order but removes duplicates) would be useful here.